src/js/glob.js

```diff
@@ -0,0 +1,349 @@
+"use strict";
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+var path      = require("path");
+var fs        = require("fs");
+var minimatch = require("minimatch");
+
+fluid.registerNamespace("gpii.glob");
+
+/**
+ *
+ * Find all files beneath a root directory based on a list of includes and excludes.  Includes and excludes can be
```

Comment:
Give a little more detail here about the exact strategy used to search. As I understand it, static includes and excludes are used to prune the search eagerly in order to avoid traversing deep hierarchies which have already been excluded.

Reply:
I fleshed out both the JSDocs a bit.